src/mut_mut.rs

```diff
@@ -27,25 +23,34 @@ impl LintPass for MutMut {
 			}
 		}
 		
-		if unwrap_addr(expr).and_then(unwrap_addr).is_some() {
-			cx.span_lint(MUT_MUT, expr.span, 
-				"We're not sure what this means, so if you know, please tell us.")
-		}
+		unwrap_addr(expr).map(|e| {
+			if unwrap_addr(e).is_some() {
+				cx.span_lint(MUT_MUT, expr.span, 
+					"We're not sure what this means, so if you know, please tell us.")
```

Comment:
Perhaps a better message? "Generally you want to avoid `&mut &mut foo` if possible"